src/components/Form/MultiSelect/MultiSelect.tsx

```diff
@@ -99,18 +99,30 @@ export const MultiSelect: FC<MultiSelectProps> = ({
 
   const filterOptions = (value: string, options: Option[]) => {
     if (value.length < 3) {
-      setFilteredOptions(options)
+      const filteredResults = options.map(option => ({
+        ...option,
+        matchesSearchPhrase: false
+      }))
+
+      setFilteredOptions(filteredResults)
 
       return
     }
 
     const lowerCaseSearch = value.toLowerCase()
-    const filteredResults = options.filter(
-      item =>
-        item.label.toLowerCase().search(lowerCaseSearch) !== -1 || item.selected
-    )
+    const searchResults = options
+      .filter(
+        option =>
+          option.label.toLowerCase().search(lowerCaseSearch) !== -1 ||
+          option.selected
+      )
+      .map(option => ({
+        ...option,
+        matchesSearchPhrase:
+          option.label.toLowerCase().search(lowerCaseSearch) !== -1
```

Comment:
może żeby nie powtarzać tego złożonego warunku, to najpierw zrobić mapowanie, a potem filtrowanie? wtedy filtrowanie będzie miało prostszą postać:
`option.selected || option.matchesSearchPhrase`

Reply:
@dbindaczlandingi bardzo trafna uwaga, ready ;>